File: WpfSamples/XBAPHostingSilverlightSample/vb/SilverlightApplication/ScriptableClass.vb
Imports System.Windows.Browser ' ScriptableMemberAttribute

Namespace SilverlightApplication
	<ScriptableType> _
	Public Class ScriptableClass
		' Method that can be called from HTML script
		<ScriptableMember> _
		Public Sub SendMessageToSilverlight(ByVal msg As String)
			msg = "Your Silverlight application has recieved a message: " & msg
			CType(App.Current.RootVisual, MainPage).msgTextBlock.Text = msg

		End Sub


	End Class
End Namespace

File: WpfSamples/XBAPHostingSilverlightSample/cs/SilverlightApplication/ScriptableClass.cs
using System.Windows.Browser; // ScriptableMemberAttribute

namespace SilverlightApplication
{
    [ScriptableType]
    public class ScriptableClass
    {
        // Method that can be called from HTML script
        [ScriptableMember]
        public void SendMessageToSilverlight(string msg)
        {
            msg = "Your Silverlight application has recieved a message: " + msg;
            ((MainPage)App.Current.RootVisual).msgTextBlock.Text = msg;
            
        }

       
    }
}
